sonic_platform_base/chassis_base.py

```diff
@@ -109,6 +109,41 @@ def get_reboot_cause(self):
         """
         raise NotImplementedError
 
+    def get_supervisor_slot(self):
+        """
+        Retrieves the slot of the supervisor-module of the modular chassis.
+        On the supervisor or line-card modules, it will return the slot of the
+        supervisor-module.
+
+        On the fixed-platforms, the API can be ignored.
+
+        Users of the API can catch the exception and return a default
+        ModuleBase.MODULE_INVALID_SLOT and bypass code for fixed-platforms.
+
+        Returns:
+            An integer, the slot identifier of the supervisor module of the
+            modular-chassis.
```

Comment:
Can you also specify whether the slot identifier is expected to be 0-based or 1-based? Please see the docstring for `get_fan()` for an example.

Reply:
This would be slightly different. This returns the physical-slot of the module in the chassis. It is vendor specific. Made changes to the comment.